File: 01384.py
group = 1
while 1:
    n = int(input())
    if not n: break

    names = []; messages = []
    for i in range(n):
        tmp = list(input().split())
        names.append(tmp[0])

        message = []
        for j in range(1, n):
            if(tmp[j] == 'N'):
                message.append(j)
        messages.append(message)

    print("Group %d" % group)
    group += 1

    isNeg = False
    for i in range(n):
        if len(messages[i]):
            for j in range(len(messages[i])):
                index = i - messages[i][j]
                if index < 0: index += n
                print(f"{names[index]} was nasty about {names[i]}")
                isNeg = True

    if not isNeg: print("Nobody was nasty")
    print()

File: 01384.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
	int group = 1;
	while (1) {
		int n; cin >> n;
		if (!n) break;

		vector<string> names;
		vector<int> messages[20];
		for (int i = 0; i < n; i++) {
			string name; cin >> name;
			names.push_back(name);

			for (int j = 1; j < n; j++) {
				char c; cin >> c;
				if (c == 'N') messages[i].push_back(j);
			}
		}

		cout << "Group " << group++ << endl;
		bool isNeg = false;
		for (int i = 0; i < n; i++) {
			if (messages[i].size()) {
				for (int j = 0; j < messages[i].size(); j++) {
					int index = i - messages[i][j];
					if (index < 0) index += n;
					cout << names[index] << " was nasty about " << names[i] << '\n';
					isNeg = true;
				}
			}
		}

		if (!isNeg) cout << "Nobody was nasty\n";
		cout << endl;
	}
}
